core/src/main/java/org/apache/iceberg/BaseOverwriteFiles.java

```diff
@@ -163,6 +163,12 @@ protected void validate(TableMetadata base, Snapshot parent) {
     }
   }
 
+  @Override
+  protected boolean useReferencedManifestsWhenFiltering(Snapshot parent) {
```

Comment:
Can we also add a comment when it is safe to use referenced manifests to explain this logic?

Reply:
I added a comment in the updated logic to use the parent manifests and perform a subset check against the referenced manifests to explain why this logic is there. If we go back to the "no intermediate snapshots" aproach, I'll add a similar comment on the implementations of this method